File: LLM-Translations/GPT-4/codenet/C/Python/s484809551.py

import sys

def strrstr(string, pattern):
    offset = 0
    lastP = None

    while True:
        searchResult = string[offset:].find(pattern)
        if searchResult != -1 and string[offset] != '\0':
            lastP = string[offset + searchResult:]
            offset += 1
        else:
            break

    return lastP

def main():
    patternBase = "keyence"
    
    S = input().strip()

    len_S = len(S)
    found = False

    for offset in range(len(patternBase)):
        top = patternBase[:offset]
        bottom = patternBase[offset:]

        topSearch = S.find(top)
        if topSearch != -1:
            bottomSearch = strrstr(S[topSearch + offset:], bottom)
            if bottomSearch is not None:
                sepCount = 0

                if topSearch != 0:
                    sepCount += 1
                if topSearch + offset != S.find(bottom, topSearch + offset):
                    sepCount += 1
                if bottomSearch[len(bottom):] != '':
                    sepCount += 1

                if sepCount <= 1:
                    found = True
                    break

    print("YES" if found else "NO")

if __name__ == "__main__":
    main()


#

File: LLM-Translations/GPT-4/codenet/C/Go/s484809551.go

package main

import (
	"fmt"
	"strings"
)

const patternBase = "keyence"

func strrstr(s, substr string) string {
	lastIndex := -1
	for {
		index := strings.Index(s, substr)
		if index == -1 {
			break
		}
		lastIndex = index
		s = s[index+1:]
	}
	if lastIndex == -1 {
		return ""
	}
	return s
}

func main() {
	var S string
	fmt.Scan(&S)

	found := false
	for offset := 0; offset < len(patternBase); offset++ {
		top := patternBase[:offset]
		bottom := patternBase[offset:]

		topSearch := strings.Index(S, top)
		if topSearch != -1 {
			bottomSearch := strings.LastIndex(S[topSearch+offset:], bottom)
			if bottomSearch != -1 {
				sepCount := 0

				if topSearch != 0 {
					sepCount++
				}
				if topSearch+offset != bottomSearch {
					sepCount++
				}
				if bottomSearch+len(bottom) != len(S) {
					sepCount++
				}

				if sepCount <= 1 {
					found = true
					break
				}
			}
		}
	}

	if found {
		fmt.Println("YES")
	} else {
		fmt.Println("NO")
	}
}


